sql-plugin/src/main/scala/org/apache/spark/sql/rapids/collectionOperations.scala

```diff
@@ -1561,6 +1563,82 @@ case class GpuFlattenArray(child: Expression) extends GpuUnaryExpression with Nu
   }
 }
 
+case class GpuArrayDistinct(child: Expression) extends GpuUnaryExpression with NullIntolerantShim {
+  private def childDataType: ArrayType = child.dataType.asInstanceOf[ArrayType]
+  override def nullable: Boolean = child.nullable || childDataType.containsNull
+  override def dataType: DataType = child.dataType
+
+  private def normalizeNaNs(input: ColumnView, elementType: DataType): ColumnVector = {
+    withResource(input.isNan()) { isNaN =>
+      val canonicalNaN = if (elementType == FloatType) {
+        Scalar.fromFloat(Float.NaN)
+      } else {
+        Scalar.fromDouble(Double.NaN)
+      }
+      withResource(canonicalNaN) { canonicalNaN =>
+        isNaN.ifElse(canonicalNaN, input)
+      }
+    }
+  }
+
+  override def doColumnar(input: GpuColumnVector): ColumnVector = {
+    val elementType = childDataType.elementType
+    elementType match {
+      case FloatType =>
+        // For Array[Float], bitCast to Array[Int32], perform distinct, then cast back
+        withResource(input.getBase.getChildColumnView(0)) { arrayElements =>
+          withResource(normalizeNaNs(arrayElements, FloatType).bitCastTo(DType.INT32))
+          { castedArrayElements =>
+            withResource(GpuListUtils.replaceListDataColumnAsView(input.getBase,
+             castedArrayElements)) { castedColumnView =>
+              withResource(castedColumnView.dropListDuplicates(DuplicateKeepOption.KEEP_FIRST))
+              { arrayDistinctColumnView =>
+                withResource(arrayDistinctColumnView.getChildColumnView(0))
+                { arrayDistinctElements =>
+                  withResource(arrayDistinctElements.bitCastTo(DType.FLOAT32))
+                  { floatArrayDistinctElements =>
+                    withResource(GpuListUtils.replaceListDataColumnAsView(arrayDistinctColumnView,
+                     floatArrayDistinctElements)) { resultView =>
+                      resultView.copyToColumnVector()
+                    }
+                  }
+                }
+              }
+            }
+          }
+        }
+      case DoubleType =>
```

Comment:
The code of both `FloatType` and `DoubleType` can be extracted as a template function.